### DISCUSSION TITLE
求助：三元表达式得到错误的结果

### DISCUSSION THREAD

**Discussion by linghuccc** [openbuildxyz/mina-zkapp-bootcamp#220] [Q&A] [answered]
![error](https://github.com/user-attachments/assets/8505ff23-1283-4e2a-9470-76e9b41ce10b)

如图，
currentApprovalNumber = 1
publicInput.vote = false
newApprovalNumber 应该是 1 才对，但是代码却得到 2，这是为什么？

**Comment by timimm** [comment, marked as answer]
试试 `Provable.if()`

**Comment by coldstar1993** [comment]
在o1js中使用Bool变量实现普通js中三元表达式的效果，要采用Provable.if(*, *, *).
问题图中的publicInput.vote是Bool类型的变量而且有对象值的，注意（在js解释器眼里）它不是js中的true!  所以如果直接用在普通js中三元表达式的话，永远是true。

**Comment by linghuccc** [comment]
收到
